lib/pos_blockchain.go

```diff
@@ -537,6 +547,29 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 	return errors.Wrap(err, "processBlockPoS: Problem adding block to block index: ")
 }
 
+// checkAndStoreArchivalBlock is a helper function that takes in a block and checks if it's an archival block.
+// If it is, it stores the block in the db and returns true. If it's not, it returns false, or false and an error.
+func (bc *Blockchain) checkAndStoreArchivalBlock(block *MsgDeSoBlock) (_success bool, _err error) {
```

Comment:
Just as a simplification, you make this work without the extra _success return value. The _err return value should be sufficient:
- if the block was successfully stored as an archival block, then _err is nil
- if the block doesn't pass the archival block conditions or there was an error, then _err is non-nil
```suggestion
func (bc *Blockchain) storeArchivalBlock(block *MsgDeSoBlock) (_err error) {
```